File: python/백준/구현/1032 명령 프롬프트.py
n = int(input())

base_string = list(input().strip())  # 문자열 → 리스트로 변환

for _ in range(n-1):
    compare_string = str(input())
    for i in range(len(base_string)):
        if base_string[i] != compare_string[i]:
            base_string[i] = "?"

print("".join(base_string))  # 리스트 → 문자열로 다시 합침    

File: cpp/백준/구현/1032 명령 프롬프트.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    string base_string;
    cin >> base_string;

    for (int k = 0; k < n - 1; ++k) {
        string compare_string;
        cin >> compare_string;
        for (int i = 0; i < (int)base_string.size(); ++i) {
            if (base_string[i] != compare_string[i]) {
                base_string[i] = '?';
            }
        }
    }

    cout << base_string << "\n";
    return 0;
}
